presto-hive/src/test/java/com/facebook/presto/hive/TestCoercingFilters.java

```diff
@@ -70,4 +79,41 @@ public void testVarcharToInteger()
 
         assertFalse(coercingFilter.testNull());
     }
+
+    private final TypeManager typeManager = MetadataManager.createTestMetadataManager().getFunctionAndTypeManager();
+
+    @Test
+    public void testShortToInteger()
+    {
+        HiveCoercer coercer = HiveCoercer.createCoercer(typeManager, HIVE_SHORT, HIVE_INT);
```

Comment:
Can we add some negative tests that the original code would fail? E.g. fromType is HIVE_DOUBLE and toType is HIVE_SHORT. Such case would create the VarcharToIntegerNumberCoercer before this fix, because the condition `toHiveType.equals(HIVE_SHORT)` on line 288 is true, but this is actually incorrect. In the test this shall throw an error and we need to verify the exception is thrown as expected.

Reply:
done